homeassistant/components/switch/enocean.py

```diff
@@ -58,23 +67,43 @@ def name(self):
 
     def turn_on(self, **kwargs):
         """Turn on the switch."""
-        optional = [0x03, ]
-        optional.extend(self.dev_id)
-        optional.extend([0xff, 0x00])
-        self.send_command(data=[0xD2, 0x01, 0x00, 0x64, 0x00,
-                                0x00, 0x00, 0x00, 0x00], optional=optional,
-                          packet_type=0x01)
-        self._on_state = True
+        if self.subtype == "" or self.subtype == "permundo":
+            optional = [0x03, ]
+            optional.extend(self.dev_id)
+            optional.extend([0xff, 0x00])
+            self.send_command(data=[0xD2, 0x01, 0x00, 0x64, 0x00,
```

Comment:
This code is a protocol implementation and should never have been allowed in Home Assistant. It needs to be extracted into a third party lib.